JoinColony/colonyCDapp
# Feat: Add payout claimed notification

## Description

Add a new notification for when a payout is claimed from an expenditure, and change the expenditure finalized notification message to instead say "Payment released".

Block ingestor changes here: https://github.com/JoinColony/block-ingestor/pull/284

## Testing

* Create an advanced payment with a few different claim delays.
<img width="1058" alt="Screenshot 2024-10-18 at 15 46 37" src="https://github.com/user-attachments/assets/2732a1f1-e0e6-465e-af2c-c57c43dadc14">

* Go through all the steps for the expenditure, until you get to the final stage (when it has been finalized). You should receive a notification about "Payment released". If you had any payments with a zero claim delay, you should also receive a notification about "Payment made".
<img width="1054" alt="Screenshot 2024-10-18 at 15 47 36" src="https://github.com/user-attachments/assets/5c0c5fe3-14b9-406f-b3a6-e3f7bcecbc04">

* Advance time to unlock some more payouts for claiming by running `npm run forward-time -- 1` in a terminal. The final number in this script is the number of hours to advance by. In my example, I advanced time by 1 hour to unlock two more payouts.
<img width="764" alt="Screenshot 2024-10-18 at 15 47 55" src="https://github.com/user-attachments/assets/12c80c97-9237-43eb-aa0d-e223832e4425">
<img width="1056" alt="Screenshot 2024-10-18 at 15 48 14" src="https://github.com/user-attachments/assets/154c4bfc-aa1f-41a6-a32b-5baedf267e9f">

* In the action sidebar, make the payments that are available and check that you received a notification for each claim.
<img width="1055" alt="Screenshot 2024-10-18 at 15 48 30" src="https://github.com/user-attachments/assets/217de0d8-38d1-43fa-8e7c-8bff99a17e47">
<img width="1058" alt="Screenshot 2024-10-18 at 15 48 38" src="https://github.com/user-attachments/assets/7a25562d-a8ad-4836-819c-f1f091e96a1b">

* Repeat the above two steps until all payouts are claimed.
<img width="735" alt="Screenshot 2024-10-18 at 15 48 51" src="https://github.com/user-attachments/assets/1b72b1a5-5eb6-4fa2-ab5d-400b55cc4f01">
<img width="1062" alt="Screenshot 2024-10-18 at 15 49 07" src="https://github.com/user-attachments/assets/adf16ff6-62cd-4584-a0ca-29ad5fb23144">
<img width="1058" alt="Screenshot 2024-10-18 at 15 49 17" src="https://github.com/user-attachments/assets/be589c38-dee7-4ab2-b681-7917b5ea1147">
<img width="1064" alt="Screenshot 2024-10-18 at 15 49 43" src="https://github.com/user-attachments/assets/fea783ce-5163-493e-87e1-caf31c14cbe7">


## Diffs

**New stuff** ✨

* New notification type `EXPENDITURE_PAYOUT_CLAIMED`

**Changes** 🏗

* Changes in the block ingestor https://github.com/JoinColony/block-ingestor/pull/284
* Changed the message for expenditure finalized notifications to "Payment released:"

Resolves #3389
